apps/studio/src/server/modules/site/site.router.ts

```diff
@@ -200,39 +200,32 @@ export const siteRouter = router({
         userId: ctx.user.id,
         action: "read",
       })
-      const notification = await getNotification(siteId)
-      return notification
+
+      return await getNotification(siteId)
     }),
   setNotification: protectedProcedure
     .input(setNotificationSchema)
-    .mutation(async ({ ctx, input }) => {
-      const { siteId, notification, notificationEnabled } = input
+    .mutation(async ({ ctx, input: { siteId, title, content, enabled } }) => {
       await validateUserPermissionsForSite({
         siteId,
         userId: ctx.user.id,
         action: "update",
       })
 
       const site = await db.transaction().execute(async (tx) => {
-        if (notificationEnabled) {
-          return await setSiteNotification({
-            tx,
-            siteId,
-            userId: ctx.user.id,
-            notification,
-          })
-        } else {
-          return await clearSiteNotification({
-            tx,
-            siteId,
-            userId: ctx.user.id,
-          })
-        }
+        return await setSiteNotification({
+          tx,
+          siteId,
+          userId: ctx.user.id,
+          title,
+          content,
+          enabled: !!enabled,
+        })
```

Comment:
ask: actually since this is the only use case of `setSiteNotification`, can we do some refactoring to move `db.transaction()` into `setSiteNotification` instead?

Reply:
sure